src/features/feed/HideReadPosts.tsx

```diff
@@ -0,0 +1,27 @@
+import { useContext } from "react";
+import { FeedContext } from "./FeedContext";
+import { IonIcon } from "@ionic/react";
+import { eyeOff } from "ionicons/icons";
+import { useAppDispatch, useAppSelector } from "../../store";
+import { hidePost } from "../post/postSlice";
+
+export default function HideReadPosts() {
+  const dispatch = useAppDispatch();
+  const { itemsRef } = useContext(FeedContext);
+  const postReadById = useAppSelector((state) => state.post.postReadById);
+
+  return (
+    <IonIcon
+      icon={eyeOff}
+      onClick={async () => {
+        const postIds: number[] = itemsRef.current.map((item) => item.post.id);
+
+        const toHide = postIds.filter((id) => postReadById[id]);
+
+        for (const postId of toHide) {
+          dispatch(hidePost(postId));
```

Comment:
@burakcan currently I do a new DB insert for every hidden post. I assume it's possible to bulk add with Dexie, but if you have any pointers that would be appreciated :) it's probably necessary with the mark read on scroll feature I'm adding to the next release

Reply:
(not a huge priority though)

Reply:
@aeharding there is a bulkadd method for tables in dexie. I guess we can debounce the add calls for a couple of frames, keep the items and then bulk add them in a single transaction. I'll look into it when I have time. Did we notice any performance issues with this?

Reply:
Actually, in terms of performance it seems totally fine in a loop :) So no need to do this, unless you want to